src/python/pants/engine/target.py

```diff
@@ -623,6 +623,15 @@ class CoarsenedTarget:
     # To expand these dependencies, request `CoarsenedTargets` for them.
     dependencies: FrozenOrderedSet[Address]
 
+    def debug_hint(self) -> str:
+        return str(self)
+
+    def __str__(self) -> str:
+        if len(self.members) > 1:
+            others = len(self.members) - 1
+            return f"{self.members[0].address.spec} (and {others} more)"
+        return self.members[0].address.spec
```

Comment:
Is this actually guaranteed to even have at least 1 member?

Reply:
It is, yea. Because a target coarsens to itself if it is not involved in any cycles.

Reply:
Ah yeah, fair enough